commit 5eba8b6150b37fb229c0f9ad54d8b555e5e3225c
Author: Vincent Untz <vuntz@opensuse.org>
Date:   Fri Feb 24 11:54:39 2012 +0100

    mgnokiidev: Respect SUID_CFLAGS and SUID_LDFLAGS
    
    This is a good way for distributors to use -fPIE/-pie when building
    gnokii.

diff --git a/configure.in b/configure.in
index ab0bb50..958b5aa 100644
--- a/configure.in
+++ b/configure.in
@@ -998,6 +998,11 @@ AC_DEFINE_UNQUOTED(SBINDIR, "$mgnokiidev_bindir", [Where mgnokiidev gets install
 AC_EXPAND_DIR(prefix_dir, "${prefix}")
 AC_DEFINE_UNQUOTED(PREFIX, "$prefix_dir", [Root dir for gnokii, xgnokii and smsd])
 
+AC_ARG_VAR([SUID_CFLAGS],
+           [CFLAGS used for binaries which are usually with the suid bit])
+AC_ARG_VAR([SUID_LDFLAGS],
+           [LDFLAGS used for binaries which are usually with the suid bit])
+
 AC_CONFIG_HEADERS([include/config.h])
 
 AC_SUBST(PACKAGE)
diff --git a/utils/Makefile.am b/utils/Makefile.am
index b28b873..763d020 100644
--- a/utils/Makefile.am
+++ b/utils/Makefile.am
@@ -3,7 +3,8 @@ DEFS = -DLOCALEDIR=\"$(localedir)\"
 sbin_PROGRAMS = mgnokiidev
 mgnokiidev_SOURCES = mgnokiidev.c
 mgnokiidev_LDADD = $(INTLLIBS) $(TCP_LIBS)
-mgnokiidev_CFLAGS = -I$(top_srcdir)/include
+mgnokiidev_LDFLAGS = $(SUID_LDFLAGS) $(AM_LDFLAGS)
+mgnokiidev_CFLAGS = -I$(top_srcdir)/include $(SUID_CFLAGS) $(AM_CFLAGS)
 
 scripts_DATA = sendsms
 scriptsdir = $(bindir)
